File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/ExportFileExplorer.java
package uk.nhs.nhsx.diagnosiskeydist;

import batchZipCreation.Exposure;
import uk.nhs.nhsx.core.aws.xray.Tracing;
import uk.nhs.nhsx.diagnosiskeydist.agspec.ENIntervalNumber;

import java.io.FileInputStream;
import java.util.Base64;
import java.util.Date;

/**
 * current?
 */
public class ExportFileExplorer {
    public static void main(String[] args) throws Exception {
        Tracing.disableXRayComplaintsForMainClasses();

        try (var in = new FileInputStream(args.length == 0 ? "export.bin" : args[0])) {
            long skipped = in.skip(DistributionService.EK_EXPORT_V1_HEADER.length());

            var export = Exposure.TemporaryExposureKeyExport.parseFrom(in);

            System.out.println("{");

            if (export.hasRegion()) {
                System.out.println("  \"region\": \"" + export.getRegion() + "\",");
            }

            if (export.hasBatchNum()) {
                System.out.println("  \"batchNum\": \"" + export.getBatchNum() + "\",");
            }

            if (export.hasBatchSize()) {
                System.out.println("  \"batchSize\": \"" + export.getBatchSize() + "\",");
            }

            if (export.hasStartTimestamp()) {
                System.out.println("  \"startTimestamp\": \"" + export.getStartTimestamp() + "\" /* " + new Date(export.getStartTimestamp() * 1000) + " */,");
            }

            if (export.hasEndTimestamp()) {
                System.out.println("  \"endTimestamp\": \"" + export.getEndTimestamp() + "\" /* " + new Date(export.getEndTimestamp() * 1000) + " */,");
            }

            System.out.println("  \"temporaryExposureKeys\": [");
            for (var i = export.getKeysList().iterator(); i.hasNext(); ) {
                var key = i.next();

                System.out.print("    {");
                System.out.print("\"key\":\"" + Base64.getEncoder().encodeToString(key.getKeyData().toByteArray()) + "\",");
                System.out.print("\"rollingStartNumber\":\"" + key.getRollingStartIntervalNumber() + "\" /* " + new ENIntervalNumber(key.getRollingStartIntervalNumber()).toTimestamp() + " */,");
                System.out.print("\"rollingPeriod\":\"" + key.getRollingPeriod() + "\" /* " + new ENIntervalNumber(key.getRollingStartIntervalNumber() + key.getRollingPeriod()).toTimestamp() + " */,");
                System.out.print("\"transmissionRiskLevel\":\"" + key.getTransmissionRiskLevel() + "\"");

                if (key.hasDaysSinceOnsetOfSymptoms()) {
                    System.out.print(",\"daysSinceOnsetOfSymptoms\":\"" + key.getDaysSinceOnsetOfSymptoms() + "\"");
                }

                if (i.hasNext()) {
                    System.out.println("},");
                }
                else {
                    System.out.println("}");
                }
            }
            System.out.println("  ]");

            System.out.println("}");
        }

        System.exit(0);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/ExportFileExplorer.kt
package uk.nhs.nhsx.diagnosiskeydist

import batchZipCreation.Exposure.TemporaryExposureKeyExport
import uk.nhs.nhsx.core.aws.xray.Tracing
import uk.nhs.nhsx.diagnosiskeydist.agspec.ENIntervalNumber
import java.io.FileInputStream
import java.util.*
import kotlin.system.exitProcess

/**
 * current?
 */
object ExportFileExplorer {
    @JvmStatic
    fun main(args: Array<String>) {
        Tracing.disableXRayComplaintsForMainClasses()
        FileInputStream(if (args.isEmpty()) "export.bin" else args[0])
            .use {
                it.skip(DistributionService.EK_EXPORT_V1_HEADER.length.toLong())
                val export = TemporaryExposureKeyExport.parseFrom(it)
                println("{")
                if (export.hasRegion()) println("""  "region": "${export.region}",""")
                if (export.hasBatchNum()) println("""  "batchNum": "${export.batchNum}",""")
                if (export.hasBatchSize()) println("""  "batchSize": "${export.batchSize}",""")
                if (export.hasStartTimestamp())
                    println("""  "startTimestamp": "${export.startTimestamp}" /* ${Date(export.startTimestamp * 1000)} */,""")
                if (export.hasEndTimestamp()) println("""  "endTimestamp": "${export.endTimestamp}" /* ${Date(export.endTimestamp * 1000)} */,""")
                println("""  "temporaryExposureKeys": [""")
                val i = export.keysList.iterator()
                while (i.hasNext()) {
                    val key = i.next()
                    print("    {")
                    print(""""key":"${Base64.getEncoder().encodeToString(key.keyData.toByteArray())}",""")
                    print(
                        """"rollingStartNumber":"${key.rollingStartIntervalNumber}" /* """ +
                            ENIntervalNumber(key.rollingStartIntervalNumber.toLong()).toTimestamp() + " */,"
                    )
                    print(""""rollingPeriod":"${key.rollingPeriod}" /* ${ENIntervalNumber((key.rollingStartIntervalNumber + key.rollingPeriod).toLong()).toTimestamp()} */,""")
                    print(""""transmissionRiskLevel":"${key.transmissionRiskLevel}"""")
                    if (key.hasDaysSinceOnsetOfSymptoms()) print(""","daysSinceOnsetOfSymptoms":"${key.daysSinceOnsetOfSymptoms}"""")
                    if (i.hasNext()) println("},") else println("}")
                }
                println("  ]")
                println("}")
            }
        exitProcess(0)
    }
}
